CanonicalLtd/discourse-markdown-note
# Add Warning, Success, titles

Fixes https://github.com/canonical-webteam/discourse-markdown-note/issues/1

This plugin is for the purposes of porting documentation (e.g. [juju-docs](https://github.com/juju/docs/)), from [documentation-builder](https://github.com/canonical-webteam/documentation-builder/) format into Discourse.

It attempts to provide a mechanism to replicate the Admonitions support [in the NotificationExtension](https://github.com/canonical-webteam/documentation-builder/blob/master/ubuntudesign/documentation_builder/extensions.py#L17):

``` python
    Usage:
        !!! Note:
            Things are reasonable
        !!! Warning "":
            I have no title
        !!! Positive "Awesome":
            Everything is now wonderful
```

However, in this plugin I try to stay closer to [the Vanilla language for notifications](https://docs.vanillaframework.io/en/patterns/notification), so I use "type" and "status". And I use BB-code, which is how Discourse Markdown extensions [tend to be implemented](https://meta.discourse.org/t/post-format-reference-documentation/19197/2). E.g.:

``` bb-code
[note]
Things are reasonable
[/note]

[note type=caution]
I have no title
[/note]

[note type=positive status="Awesome"]
Everything is now wonderful
[/note]
```

QA
--

This basically replicates [our instructions for running Discourse](https://github.com/canonical-webteam/practices/wiki/Running-Discourse-with-Docker-for-local-development):

``` bash
git clone https://github.com:discourse/discourse.git
cd discourse/plugins
git clone -b warning-success-titles https://github.com:nottrobin/discourse-markdown-note
cd ..
./bin/docker/boot_dev --init
./bin/docker/rails s
```

Once you've got the server running, go to http://localhost:3000, and log in with the admin user you just created.

Now try to create a post with the contents:

``` bash
[note]
Things are reasonable
[/note]

[note type=caution]
I have no title
[/note]

[note type=positive status="Awesome"]
Everything is now wonderful
[/note]
```

Although the styling within Discourse will be basic, if you inspect the markup that's generated, either in the editor or in the actual post once you save it, it should match the [Vanilla notification pattern](https://docs.vanillaframework.io/en/patterns/notification) markup format.

This means that once the markup is imported from Discourse to be displayed on a website running Vanilla, it will display as a Vanilla notification.
